packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -541,57 +541,12 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
    * Sets the connected account
    *
    * `AccountInfo` type comes from a legacy wallet adapter plugin
-   * `StandardAccountInfo` type comes from AIP-62 standard compatible wallet when onAccountChange event is called
-   * `UserResponse<StandardAccountInfo>` type comes from AIP-62 standard compatible wallet on wallet connect
+   * `StandardAccountInfo` type comes from AIP-62 standard compatible wallet
    *
    * @param account An account
    */
-  setAccount(
-    account:
-      | AccountInfo
-      | StandardAccountInfo
-      | UserResponse<StandardAccountInfo>
-      | null
-  ): void {
-    if (account === null) {
-      this._account = null;
-      return;
-    }
-
-    // Check if wallet is of type AIP-62 standard
-    if (this._wallet?.isAIP62Standard) {
-      // Check if account is of type UserResponse<StandardAccountInfo> which means the `account`
-      // comes from the `connect` method
-      if ("status" in account) {
-        const connectStandardAccount =
-          account as UserResponse<StandardAccountInfo>;
-        if (connectStandardAccount.status === UserResponseStatus.REJECTED) {
-          this._connecting = false;
-          throw new WalletConnectionError("User has rejected the request")
-            .message;
-        }
-        // account is of type
-        this._account = {
-          address: connectStandardAccount.args.address.toString(),
-          publicKey: connectStandardAccount.args.publicKey.toString(),
-          ansName: connectStandardAccount.args.ansName,
-        };
-        return;
-      } else {
-        // account is of type `StandardAccountInfo` which means it comes from onAccountChange event
-        const standardAccount = account as StandardAccountInfo;
-        this._account = {
-          address: standardAccount.address.toString(),
-          publicKey: standardAccount.publicKey.toString(),
-          ansName: standardAccount.ansName,
-        };
-        return;
-      }
-    }
-
-    // account is of type `AccountInfo`
-    this._account = { ...(account as AccountInfo) };
-    return;
+  setAccount(account: AccountInfo | StandardAccountInfo | null): void {
+    this._account = account;
   }
```

Comment:
This would be a good spot where we support both `AccountInfo`s (the old internal type, and the standard type) as inputs, and normalize it to the standard one.